localstack-core/localstack/utils/lambda_debug_mode/lambda_debug_mode_config.py

```diff
@@ -0,0 +1,198 @@
+from __future__ import annotations
+
+import logging
+from typing import Optional
+
+import yaml
+from pydantic import BaseModel, Field, ValidationError
+from yaml import Loader, MappingNode, MarkedYAMLError, SafeLoader
+
+from localstack.aws.api.lambda_ import Arn
+
+LOG = logging.getLogger(__name__)
+
+
+class LambdaDebugConfig(BaseModel):
+    debug_port: Optional[int] = Field(None, alias="debug-port")
+    timeout_disable: bool = Field(False, alias="timeout-disable")
```

Comment:
I'm wondering about the naming: Would `enforce_timeouts: True|False` be clearer? What's your naming suggestion @dfangl @dominikschubert ?

Current behavior:
* `LAMBDA_DEBUG_MODE=0` -> never disable any timeout
* `LAMBDA_DEBUG_MODE=1`
  * `timeout-disable` not set or no debug config for Lambda function ARN available -> extend timeouts
  * `timeout-disable=False` -> extend timeouts
  * `timeout-disable=True` -> enforce timeouts as expected

Am I getting confused here or is this a logic error (the default value is `False`, which gets negated in `is_lambda_debug_timeout_for`) 🤔

sidenote: Technically, we're not fully disabling the timeouts, but extend them.


We need to document the behavior clearly after merging ...